crates/chain/src/tx_graph.rs

```diff
@@ -319,6 +319,12 @@ impl<A> TxGraph<A> {
             .map(|(outpoint, spends)| (outpoint.vout, spends))
     }
 
+    /// Creates an iterator of ancestor transactions from the starting `Transaction`, up to a
+    /// specified depth limit.
+    pub fn walk_ancestors(&self, tx: Transaction, depth_limit: usize) -> TxAncestors<A> {
```

Comment:
```suggestion
    pub fn walk_ancestors<'g>(&'g self, tx: &'g Transaction, depth_limit: usize) -> TxAncestors<'g, A> {
```

also make `TxAncestors` iterate references to `Transaction`s.